packages/@aws-cdk/cdk/lib/cloudformation/intrinsics.ts

```diff
@@ -0,0 +1,27 @@
+import { ProvisioningEngine, StringFragment } from "../core/engine-strings";
+import { IntrinsicToken  } from "../core/tokens";
+
+/**
+ * The default intrinsics Token engine for CloudFormation
+ */
+export const CLOUDFORMATION_ENGINE = 'cloudformation';
+
+/**
+ * Base class for CloudFormation built-ins
+ */
+export class CloudFormationIntrinsicToken extends IntrinsicToken {
+    protected readonly engine: string = CLOUDFORMATION_ENGINE;
```

Comment:
Instead of using a string value for `engine`, why not just point to a token resolver function?

Reply:
It's not a `resolver` function, it's a `join strings together in the engine` function. Feels overly specific to me for intrinsics to carry around a function pointer to that.

But I guess I'm with ya on them having a direct reference to the engine, instead of indirected through a string pointing to a table.

Reply:
`joiner`?